crates/astria-sequencer/src/app.rs

```diff
@@ -415,7 +436,68 @@ impl App {
             );
         }
 
-        (signed_txs, validated_txs)
+        Ok((signed_txs, validated_txs))
+    }
+
+    /// Executes the given transaction data, writing it to the app's `StateDelta`
+    /// and stores the results in `self.execution_result`.
+    ///
+    /// Will throw error if any transaction fails to decode or execute as these
+    /// transactions should've been filtered in prepare proposal.
+    ///
+    /// Returns the transactions which were [`SignedTransaction`] form.
+    #[instrument(name = "App::execute_block_data", skip_all, fields(
+        tx_count = txs.len()
+    ))]
+    async fn execute_block_data(
+        &mut self,
+        txs: VecDeque<bytes::Bytes>,
+    ) -> anyhow::Result<Vec<SignedTransaction>, anyhow::Error> {
+        let mut signed_txs = Vec::new();
+        for tx in txs {
+            // all txs in the proposal should be deserializable and executable,
+            // if any txs were not deserializeable or executable, they would not have been
+            // returned by `execute_and_filter_block_data`
+            let signed_tx = signed_transaction_from_bytes(&tx).context(
+                "failed to decode tx bytes to signed transaction, should've been filtered out in \
+                 prepare proposal",
+            )?;
+
+            // execute transaction and ensure that it succeeded
+            let tx_hash = self.execute_and_store_signed_tx(&signed_tx).await;
+            if let Some(Err(_e)) = self.execution_result.get(&tx_hash) {
+                bail!(
+                    "transaction failed during execution, should've been filtered out in prepare \
+                     proposal"
+                )
```

Comment:
should also remove the tx from the execution_results map as this will cause a protocol error, but it's already a protocol error if a tx in a finalized block failed :/ so this is fine, just gonna note this for when i merge with my branch later